appeals/web/src/server/appeals/appeal-details/assign-user/assign-user.controller.js

```diff
@@ -0,0 +1,377 @@
+import config from '#environment/config.js';
+import { setAppealAssignee } from './assign-user.service.js';
+import { mapAssignedUserToSummaryListBuilderParameters } from './assign-user.mapper.js';
+import { generateSummaryList } from '#lib/nunjucks-template-builders/summary-list-builder.js';
+
+/**
+ *
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ * @param {boolean} [isInspector]
+ * @param {Object<string, any>[]} [usersMatchingSearchTerm]
+ */
+const renderAssignUser = async (
+	request,
+	response,
+	isInspector = false,
+	usersMatchingSearchTerm
+) => {
+	const {
+		errors,
+		body: { searchTerm }
+	} = request;
+
+	const appealDetails = await appealDetailsService
+		.getAppealDetailsFromId(request.apiClient, request.params.appealId)
+		.catch((error) => logger.error(error));
+
+	const appealReferenceFragments = appealDetails?.appealReference.split('/');
+
+	if (appealDetails) {
+		const results = (usersMatchingSearchTerm || []).map((user) => {
+			const nameFragments = user.name.split(',');
+
+			return {
+				guid: user.id,
+				name:
+					nameFragments.length > 1 ? `${nameFragments[1]} ${nameFragments[0]}` : nameFragments[0],
+				email: user.email
+			};
+		});
+
+		const currentlyAssignedUser = await mapAssignedUserToSummaryListBuilderParameters(
+			appealDetails[isInspector ? 'inspector' : 'caseOfficer'],
+			appealDetails.appealId,
+			isInspector,
+			request.session
+		);
+
+		let currentAssignee;
+		if (currentlyAssignedUser) {
+			currentAssignee = generateSummaryList(currentlyAssignedUser);
+		}
+
+		return response.render('appeals/appeal/assign-user.njk', {
+			appeal: {
+				id: appealDetails.appealId,
+				reference: appealDetails.appealReference,
+				shortReference: appealReferenceFragments?.[appealReferenceFragments.length - 1]
+			},
+			isInspector,
+			currentAssignee,
+			search: {
+				term: searchTerm || '',
+				performed: Array.isArray(usersMatchingSearchTerm),
+				results
+			},
+			errors
+		});
+	}
+
+	return response.render('app/404.njk');
+};
+
+/**
+ *
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ * @param {boolean} [isInspector]
+ * @param {boolean} [isUnassign]
+ */
+const renderAssignOrUnassignUserCheckAndConfirm = async (
+	request,
+	response,
+	isInspector = false,
+	isUnassign = false
+) => {
+	const appealDetails = await appealDetailsService
+		.getAppealDetailsFromId(request.apiClient, request.params.appealId)
+		.catch((error) => logger.error(error));
+
+	const appealReferenceFragments = appealDetails?.appealReference.split('/');
+
+	if (appealDetails) {
+		const {
+			errors,
+			params: { assigneeId }
+		} = request;
+
+		if (assigneeId) {
+			const user = await usersService.getUserByRoleAndId(
```

Comment:
It doesn't look like this is dependant on appealsDetails being fetched. Can it be done in parallel?